packages/jazz-tools/src/react-core/hooks.ts

```diff
@@ -33,40 +31,58 @@ import {
   getUnloadedCoValueWithoutId,
   type BranchDefinition,
 } from "jazz-tools";
-import { JazzContext, JazzContextManagerContext } from "./provider.js";
+import { JazzContext } from "./provider.js";
 import { getCurrentAccountFromContextManager } from "./utils.js";
 import { CoValueSubscription } from "./types.js";
 import { use } from "./use.js";
 
+export function useJazzProviderCheck() {
+  const contextManager = useContext(JazzContext);
+
+  if (contextManager) {
+    throw new Error(
+      "You can't nest a JazzProvider inside another JazzProvider.",
+    );
+  }
+}
+
 export function useJazzContext<Acc extends Account>() {
-  const value = useContext(JazzContext) as JazzContextType<Acc>;
+  const value = useContext(JazzContext) as JazzContextManager<Acc, {}>;
 
   if (!value) {
     throw new Error(
-      "You need to set up a JazzProvider on top of your app to use this hook.",
+      "You need to set up a JazzProvider on top of your app to use this hook",
     );
   }
 
   return value;
 }
 
-export function useJazzContextManager<Acc extends Account>() {
-  const value = useContext(JazzContextManagerContext) as JazzContextManager<
-    Acc,
-    {}
-  >;
+export function useJazzContextValue<Acc extends Account>() {
+  const contextManager = useJazzContext<Acc>();
 
-  if (!value) {
+  const context = useSyncExternalStore(
+    useCallback(
+      (callback) => {
+        return contextManager.subscribe(callback);
+      },
+      [contextManager],
+    ),
+    () => contextManager.getCurrentValue(),
+    () => contextManager.getCurrentValue(),
+  );
```

Comment:
Re-rendering wise it is better to subscribe here instead of having JazzProvider do the subscription